plat/kvm/efi.c

```diff
@@ -340,7 +350,22 @@ static void uk_efi_setup_bootinfo_mrds(struct ukplat_bootinfo *bi)
 		uk_efi_crash("Failed to free rt_mrds\n");
 
 	/* Get memory map through GetMemoryMap */
-	uk_efi_get_mmap(&map_start, &map_sz, &desc_sz);
+	uk_efi_get_mmap(&uk_efi_map_key, &map_start, &map_sz, &desc_sz);
+
+	/* We now exit Boot Services since we no longer need it */
+	/* In case of exit failure, we obtain the memory map again, since the
+	 * memory map may have been changed
+	 */
+	status = uk_efi_bs->exit_boot_services(uk_efi_sh, uk_efi_map_key);
+	if (unlikely(status != UK_EFI_SUCCESS)) {
+		uk_efi_get_mmap(&uk_efi_map_key, &map_start, &map_sz, &desc_sz);
+		status =
+		    uk_efi_bs->exit_boot_services(uk_efi_sh, uk_efi_map_key);
+		if (unlikely(status != UK_EFI_SUCCESS))
+			uk_efi_crash(
+			    "Failed to exit Boot Services second time\n");
```

Comment:
Ditto, but keep the string on the same line, even if it is longer than 80 characters so that we can grep for it afterwards if we have it as an error message.
```C
uk_efi_crash("Failed to exit Boot Services second time\n");
```